### PR TITLE
Exception rules modification

### PR DESCRIPTION
**Pull request by schalkms**

### CODE REVIEW

File: detekt-cli/src/main/resources/default-detekt-config.yml [hunk 1]
```diff
@@ -103,7 +103,9 @@ exceptions:
       - Throwable
   InstanceOfCheckForException:
     active: false
-  PrintExceptionStackTrace:
+  InstantiateIllegalArgumentExceptionCorrectly:
```
Comment (vanniktech) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: Don't you mean?

```diff
-InstantiateIllegalArgumentExceptionCorrectly
+InstantiateIllegalArgumentExceptionInCorrectly
```

Comment (schalkms) [reply detekt-cli/src/main/resources/default-detekt-config.yml]: Thanks for the review. Yes, that makes more sense.

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/InstantiateIllegalArgumentExceptionIncorrectly.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+class InstantiateIllegalArgumentExceptionIncorrectly(config: Config = Config.empty) : Rule(config) {
+
+	override val issue = Issue("InstantiateIllegalArgumentExceptionIncorrectly", Severity.Warning,
+			"A call to the default constructor of an IllegalArgumentException was detected. " +
+					"Instead one of the constructor overloads should be called. " +
+					"This allows to provide more meaningful exceptions.",
+			Debt.FIVE_MINS)
+
+	override fun visitCallExpression(expression: KtCallExpression) {
+		if (expression.calleeExpression?.text == "IllegalArgumentException" && expression.valueArguments.isEmpty()) {
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/InstantiateIllegalArgumentExceptionIncorrectly.kt]: Should we make this configurable? Throwing exceptions without message or cause makes maintanability harder or something :) ?

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/InstantiateIllegalArgumentExceptionIncorrectly.kt]: Why not?

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/InstantiateIllegalArgumentExceptionIncorrectly.kt]: Do you have ideas for other build-in exceptions where this rule is applicable.

Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/InstantiateIllegalArgumentExceptionIncorrectly.kt]: I would just say all the exceptions that are often used like `Runtime-, IllegalArgument-, IllegalState-, IOExpection`. We do not need to make the exceptions exhaustive.

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/InstantiateIllegalArgumentExceptionIncorrectly.kt]: Ok. Then I have to rename the rule. What name to you suggest?

Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/InstantiateIllegalArgumentExceptionIncorrectly.kt]: ThrowingExceptionsWithoutMessageOrCause ? :D 
ExceptionsFromDefaultConstructor?

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/InstantiateIllegalArgumentExceptionIncorrectly.kt]: I like the first one.

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ThrowingExceptionsWithoutMessageOrCause.kt [hunk 3]
```diff
@@ -9,19 +9,28 @@ import io.gitlab.arturbosch.detekt.api.Rule
...
+	override val issue = Issue("ThrowingExceptionsWithoutMessageOrCause", Severity.Warning,
+			"A call to the default constructor of an exception was detected. " +
 					"Instead one of the constructor overloads should be called. " +
 					"This allows to provide more meaningful exceptions.",
 			Debt.FIVE_MINS)
 
+	private val exceptions: List<String> =
+			valueOrDefault(EXCEPTIONS, "IllegalArgumentException,IllegalStateException,IOException")
+			.split(',')
+			.filter { it.isNotBlank() }
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ThrowingExceptionsWithoutMessageOrCause.kt]: Take a look at the `Excludes` class. That a common behavior we introduced a own class for :)

Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ThrowingExceptionsWithoutMessageOrCause.kt]: Ups, forgot about this and merged it :X

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/ThrowingExceptionsWithoutMessageOrCause.kt]: No worries. I just create a new PR.

**Pull request merged**